tests/performance/datasets/categorical.py

```diff
@@ -0,0 +1,136 @@
+"""Dataset Generators for categorical transformers."""
```

Comment:
Should we add a dataset that produces a mixture of types?
We should also think about adding other types that can be used as categories, like `datetimes`, `floats` or `bools`.
Maybe it is not necessary to add explicitly datasets for each possible type, but at least have one dataset that produces a mixture of all of them, as well as NaNs and Nones.